draft-ietf-tls-tls13.md

```diff
@@ -3872,91 +3885,35 @@ and renegotiation_info {{RFC5746}}.
-| TLS_DHE_RSA_WITH_AES_256_CCM                  | {0xC0,0x9F} | [RFC6655]     |
-| TLS_DHE_RSA_WITH_AES_128_CCM_8                | {0xC0,0xA2} | [RFC6655]     |
-| TLS_DHE_RSA_WITH_AES_256_CCM_8                | {0xC0,0xA3} | [RFC6655]     |
-| TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256   | {0xCC,0xA8}   | [RFC7905] |
-| TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 | {0xCC,0xA9}   | [RFC7905] |
-| TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256     | {0xCC,0xAA}   | [RFC7905] |
-
-Note: ECDHE AES GCM was not yet standards track prior to the publication of
-this specification. This document promotes the above-listed ciphers to
-standards track.
-
-The following is a list of standards track ephemeral pre-shared key cipher
-suites which are currently available in TLS 1.3:
-
-|              Cipher Suite Name                |    Value    | Specification |
-|:----------------------------------------------|:------------|:--------------|
-| TLS_DHE_PSK_WITH_AES_128_GCM_SHA256           | {0x00,0xAA} | [RFC5487]     |
-| TLS_DHE_PSK_WITH_AES_256_GCM_SHA384           | {0x00,0xAB} | [RFC5487]     |
-| TLS_DHE_PSK_WITH_AES_128_CCM                  | {0xC0,0xA6} | [RFC6655]     |
-| TLS_DHE_PSK_WITH_AES_256_CCM                  | {0xC0,0xA7} | [RFC6655]     |
-| TLS_PSK_DHE_WITH_AES_128_CCM_8                | {0xC0,0xAA} | [RFC6655]     |
-| TLS_PSK_DHE_WITH_AES_256_CCM_8                | {0xC0,0xAB} | [RFC6655]     |
-| TLS_ECDHE_PSK_WITH_AES_128_GCM_SHA256         | {0xD0,0x01}   | [I-D.mattsson-tls-ecdhe-psk-aead] |
-| TLS_ECDHE_PSK_WITH_AES_256_GCM_SHA384         | {0xD0,0x02}   | [I-D.mattsson-tls-ecdhe-psk-aead] |
-| TLS_ECDHE_PSK_WITH_AES_128_CCM_8_SHA256       | {0xD0,0x03}   | [I-D.mattsson-tls-ecdhe-psk-aead] |
-| TLS_ECDHE_PSK_WITH_AES_128_CCM_SHA256         | {0xD0,0x04}   | [I-D.mattsson-tls-ecdhe-psk-aead] |
-| TLS_ECDHE_PSK_WITH_AES_256_CCM_SHA384         | {0xD0,0x05}   | [I-D.mattsson-tls-ecdhe-psk-aead] |
-| TLS_ECDHE_PSK_WITH_CHACHA20_POLY1305_SHA256   | {0xCC,0xAC}   | [RFC7905] |
-| TLS_DHE_PSK_WITH_CHACHA20_POLY1305_SHA256     | {0xCC,0xAD}   | [RFC7905] |
-
-Note: The values listed for ECDHE PSK AES suites are preliminary but
-are being or will be used for interop testing and therefore are likely to be
-assigned.
-
-Note: [RFC6655] is inconsistent with respect to the ordering of components
-within PSK AES CCM cipher suite names. The names above are as defined.
-
-All cipher suites in this section are specified for use with both TLS 1.2
-and TLS 1.3, as well as the corresponding versions of DTLS.
-(see {{backward-compatibility}})
+represents 256-bit AES in the GCM mode of operation. 
+
+|       Cipher Suite Name         |    Value    | Specification |
+|:--------------------------------|:------------|:--------------|
+| TLS_AES_128_GCM_SHA256        | {0x13,0x01} |   [This RFC]  |
+| TLS_AES_256_GCM_SHA384        | {0x13,0x02} |   [This RFC]  |
+| TLS_CHACHA20_POLY1305_SHA256  | {0x13,0x03} |   [This RFC]  |
```

Comment:
On Thu, Aug 11, 2016 at 5:56 PM, Dave Garrett <notifications@github.com>
wrote:

> In draft-ietf-tls-tls13.md
> <https://github.com/tlswg/tls13-spec/pull/559#discussion_r74529138>:
>
> > -are being or will be used for interop testing and therefore are likely to be
> > -assigned.
> > -
> > -Note: [RFC6655] is inconsistent with respect to the ordering of components
> > -within PSK AES CCM cipher suite names. The names above are as defined.
> > -
> > -All cipher suites in this section are specified for use with both TLS 1.2
> > -and TLS 1.3, as well as the corresponding versions of DTLS.
> > -(see {{backward-compatibility}})
> > +represents 256-bit AES in the GCM mode of operation.
> > +
> > +|       Cipher Suite Name         |    Value    | Specification |
> > +|:--------------------------------|:------------|:--------------|
> > +| TLS_AES_128_GCM_SHA256        | {0x13,0x01} |   [This RFC]  |
> > +| TLS_AES_256_GCM_SHA384        | {0x13,0x02} |   [This RFC]  |
> > +| TLS_CHACHA20_POLY1305_SHA256  | {0x13,0x03} |   [This RFC]  |
>
> Well, using these new cipher suite codepoints is required to use these
> ciphers with TLS 1.3+, so not having one for a cipher that was compatible
> before this rewrite is equivalent to banning said cipher.
>
We're no more banning it than we are banning CAMELLIA, which also can in
principle be used with TLS 1.3 but which we're not defining a code point
for. The relevant question is whether it's desirable to define a code point
for CCM in *this* document.

I am stating no opinion on whether CCM should be used, in general, however
> it is listed as standards track and was previously in the list here, so I
> don't see a reason it should be dropped in the new system. If we are to
> change support, I would recommend it be done in a separate PR, as this is
> supposed to be a negotiation change here. I just think not having a
> codepoint for it here requires a justification, not lacking one. Probably
> warrants a mailing list discussion.
>



> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tlswg/tls13-spec/pull/559/files/d328db654998facb0c96b0c03b3df238a735b4db#r74529138>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABD1oR3gNTIMgjAgmWzOLdx0TjdjlH8jks5qe8SzgaJpZM4JUEcw>
> .
>